snap/snapcraft.yaml

```diff
@@ -0,0 +1,228 @@
+name: yubioath-desktop
+adopt-info: yubioath-desktop
+icon: resources/icons/com.yubico.yubioath.svg
+base: core18
+grade: devel
+confinement: strict
+
+plugs:
+  ykman-config:
+    interface: personal-files
+    write:
+      - $HOME/.ykman
+apps:
+  pcscd:
+    command: usr/sbin/pcscd --foreground --auto-exit
+    daemon: simple
+    common-id: com.yubico.pcscd
+    plugs:
+      - hardware-observe
+      - network
+      - network-bind
+      - raw-usb
+
+  yubioath-desktop:
+    adapter: full
+    command: yubioath-desktop
+    common-id: com.yubico.yubioath
+    environment:
+      LD_LIBRARY_PATH: $SNAP/opt/qt514/lib/x86_64-linux-gnu/:$SNAP/opt/qt514/lib/:$LD_LIBRARY_PATH:$SNAP/usr/lib/
+      QT_QPA_PLATFORM: xcb
+      QT_BASE_DIR: $SNAP/opt/qt514/
+      QT_PLUGIN_PATH: $SNAP/opt/qt514/plugins/
+      QML2_IMPORT_PATH: $SNAP/opt/qt514/qml/
+    plugs:
+      - desktop
+      - desktop-legacy
+      - wayland
+      - unity7
+      - opengl
+      - x11
+      - raw-usb
+      - u2f-devices
+      - hidraw
+      - hardware-observe
+      - ykman-config
+    extensions: [gnome-3-28]
```

Comment:
@dagheyman why qt app needs gnome extension?

Reply:
It's a good question, I will investigate if it works with the `kde-neon` extension instead. Thanks!

Reply:
According to https://snapcraft.io/docs/kde-neon-extension :

> This extension is currently only available for the KDE project itself. Snaps from non-KDE publishers will not get automatically connected to the KDE runtime snap required by this extension.

However if you build your own qt then maybe you don't need any extension?

Reply:
Ah, that's probably why I didn't added it initially. Adding the gnome-extension definitely does something that is needed, but I need to investigate exactly what it does and if we could do the same without it.

Reply:
Doc says it connects to [gtk-common-themes](https://snapcraft.io/gtk-common-themes) and [gnome-3-28-1804](https://snapcraft.io/gnome-3-28-1804) content snaps.  You may specify the former manually and the latter shouldn't be needed.

The kde-neon extension docs say:

> if your application needs access to the Qt5 and KDE Frameworks development tools, add kde-frameworks-5-core18-sdk to the build-snaps of the part that builds your application. Some applications also require the Qt runtime libraries to be present at build time. In that case, also add kde-frameworks-5-core18 to build-snaps.

So you may specify `gtk-common-themes` and `kde-frameworks-5-core18` instead of using `gnome-3-28-1804` extension.

Extensions also sets:

> 
>     desktop
>     desktop-legacy
>     wayland
>     x11
> 

but you set above explicitly in manifest anyway.

Reply:
They also recommend:

> For a complete picture of what this extension does, add it to your app definition and run snapcraft expand-extensions.

Reply:
Thanks for the pointers.

However I cant seem to get this to work without the gnome-extension currently, I'm getting all kinds of errors related to openGL and fonts without it. So I'm leaving it in for now.

Pull requests welcome though!

Reply:
I'm not snap expert neither. I think if you post your manifest as _Call for testing_ you would get more eyeballs on this: https://forum.snapcraft.io/c/snap

Reply:
Good idea! Might do that later actually, thanks.